app/transformations/ETMPToFrontEnd/MemberEventReport.scala

```diff
@@ -16,31 +16,265 @@
 
 package transformations.ETMPToFrontEnd
 
-import transformations.Transformer
+import models.enumeration.EventType
+import models.enumeration.EventType.{Event2, Event3, Event4, Event5, Event6, Event7, Event8, Event8A}
 import play.api.libs.functional.syntax._
 import play.api.libs.json.Reads._
 import play.api.libs.json._
+import transformations.Transformer
+
+
+//noinspection ScalaStyle
+object MemberEventReport {
+
+  import transformations.ETMPToFrontEnd.Paths._
+  import transformations.ETMPToFrontEnd.ReadsUtilities._
+
+  def rds1832Api(eventType: EventType): Reads[JsObject] = pathUaEventDetailsForEventType(eventType).json.copyFrom(pathEtmpEventDetails.read(readsMembers(eventType)))
+
+  private def readsMembers(eventType: EventType): Reads[JsArray] = __.read(Reads.seq(readsMemberDetailsByEventType(eventType))).map(JsArray(_))
+
+  private def readsMemberDetailsByEventType(eventType: EventType): Reads[JsObject] = eventType match {
+    case Event2 =>          rdsMemberDetailsEvent2
+    case Event3 =>          rdsMemberDetailsEvent3
+    case Event4 =>          rdsMemberDetailsEvent4
+    case Event5 =>          rdsMemberDetailsEvent5
+    case Event6 =>          rdsMemberDetailsEvent6
+    case Event7 =>          rdsMemberDetailsEvent7
+    case Event8 =>          rdsMemberDetailsEvent8
+    case Event8A =>         rdsMemberDetailsEvent8A
+    case _ =>               rdsMemberDetailsEvent22And23
+  }
+
+  implicit val rdsMemberDetailsEvent2:        Reads[JsObject] = (readsDeceasedMemberDetails and readsBeneficiaryDetails and readsEvent2PaymentDetails).reduce
```

Comment:
I don't think these need to be implicit. They should be private too I think

Reply:
Addressed here: https://github.com/hmrc/pension-scheme-event-reporting/pull/74/commits/7192c98d8dcdc93ebebf3355c39e0b712a2a81f6